R/create_query_data.R

```diff
@@ -0,0 +1,894 @@
+#'
+#'   - `sdg_select`: A `sdg_select()` object.
+#'   - `version`: The WHO drug dictionary version. The value specified for the
+#'   `whodd_version` in the `create_query_data()` call is passed to this
+#'   parameter.
+#'   - `keep_id`: If set to `TRUE`, the output dataset must contain the
+#'   `QUERY_ID` variable
+#'   - `temp_env`: A temporary environment is passed to this parameter. It can
+#'   be used to store data which is used for all SDGs in the
+#'   `create_query_data()` call. For example if the SDGs need to be read from a
+#'   database all SDGs can be read and stored in the environment when the first
+#'   SDG is handled. For the other SDGs the terms can be retrieved from the
+#'   environment instead of accessing the database again.
+#'
+#' @details For each query the terms belonging to the query are determined with
+#'   respect to the `definition` element.
+#'   * If it is a `smq_select()` object, the terms are read from the SMQ
+#'   database by calling the function specified for the `get_smq_fun` parameter.
+#'   * If it is a `sdg_select()` object, the terms are read from the SDG
+#'   database by calling the function specified for the `get_sdg_fun` parameter.
+#'   * If it is a data frame, the terms stored in the data frame are used.
+#'   * If it is a list of data frames and `smq_select()` objects, all terms from
+#'   the dataframes and all terms read from the SMQ database referrenced by the
+#'   `smq_select()` objects are collated.
+#'
+#'   The following variables are created:
+#'
+#'   * `VAR_PREFIX`: Prefix of the variables to be created by
+#'   `derive_vars_query()` as specified by the `prefix` element.
+#'   * `QUERY_NAME`: Name of the query as specified by the `name` element.
+#'   * `QUERY_ID`: Id of the query as specified by the `id` element. If the `id`
+#'   element is not specified for a query, the variable is set to `NA`. If the
+#'   `id` element is not specified for any query, the variable is not created.
+#'   * `QUERY_SCOPE`: scope of the query as specified by the `scope` element of
+#'   the `smq_select()` object. For queries not defined by a `smq_select()`
+#'   object, the variable is set to `NA`. If none of the queries is defined by a
+#'   `smq_select()` object, the variable is not created.
+#'   * `QUERY_SCOPE_NUM`: numeric scope of the query. It is set to `1` if the
+#'   scope is broad. Otherwise it is set to '2'. If the `add_scope_num` element
+#'   equals `FALSE`, the variable is set to `NA`. If the `add_scope_num` element
+#'   equals `FALSE` for all SMQs or none of the queries is a SMQ , the variable
+#'   is not created.
+#'   * `TERM_LEVEL`: Name of the variable used to identify the terms.
+#'   * `TERM_NAME`: Value of the term variable if it is a character variable.
+#'   * `TERM_ID`: Value of the term variable if it is a numeric variable.
+#'
+#' @author Stefan Bundfuss
+#'
+#' @return A dataset to be used as input for `derive_vars_query()`
```

Comment:
I have added the `dataset_queries` argument.

I am not sure whether we should use dataframe or dataset. I think we are not really consistent but in most cases we use dataset instead of dataframe.